tools/hangar/__snapshots__/test_corpus/valid/inflight_class_capture_preflight_object.test.w_compile_tf-aws.md

```diff
@@ -40,3 +546,245 @@ $APP.synth();
 //# sourceMappingURL=preflight.cjs.map
 ```
 
+## preflight.file1-3.cjs
+```cjs
+"use strict";
+const $stdlib = require('@winglang/sdk');
+const std = $stdlib.std;
+const $helpers = $stdlib.helpers;
+const $extern = $helpers.createExternRequire(__dirname);
+let $preflightTypesMap = {};
+const blah = require("./preflight.inner-2.cjs");
+Object.assign($preflightTypesMap, blah.$preflightTypesMap);
```

Comment:
Is it possible for assignments to shadow each other? It might be worth asserting against this at runtime somehow just to prevent future bugs for when other are trying to make changes to this code in the future

Reply:
This should never happen due to the type unique ids.
:+1: Added a runtime check for this.